.github/workflows/ci.yaml

```
@@ -3,120 +3,95 @@
-      fail-fast: false
-      matrix:
-        tox-environments:
-          - charm-integration
-          - ha-integration
-          - tls-integration
-          - metrics-integration
-    name: ${{ matrix.tox-environments }}
+    name: Integration test charm | 3.1.7
     needs:
       - lint
       - unit-test
       - build
-    runs-on: ubuntu-latest
-    timeout-minutes: 120
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Setup operator environment
-        # TODO: Replace with custom image on self-hosted runner
-        uses: charmed-kubernetes/actions-operator@main
-        with:
-          provider: microk8s
-          channel: "1.27-strict/stable"
-          bootstrap-constraints: "cores=2 mem=2G"
-          juju-channel: 3.1/stable
-          bootstrap-options: "--agent-version 3.1.6"
-      - name: Download packed charm(s)
-        uses: actions/download-artifact@v3
-        with:
-          name: ${{ needs.build.outputs.artifact-name }}
-      - name: Check juju version
-        run: |
-          echo $(juju --version)
-      - name: Free disk space
-        run: |
-          echo "Free disk space before cleanup"
-          df -T
-          # free space in the runner
-          sudo rm -rf /usr/share/dotnet
-          sudo rm -rf /opt/ghc
-          sudo rm -rf /usr/local/share/boost
-          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
-          echo "Free disk space after cleanup"
-          df -T
-      - name: Select tests
-        id: select-tests
-        run: |
-          if [ "${{ github.event_name }}" == "schedule" ]
-          then
-            echo Running unstable and stable tests
-            echo "mark_expression=" >> $GITHUB_OUTPUT
-          else
-            echo Skipping unstable tests
-            echo "mark_expression=not unstable" >> $GITHUB_OUTPUT
-          fi
-      - name: Run integration tests
-        # set a predictable model name so it can be consumed by charm-logdump-action
-        run: sg snap_microk8s -c "tox run -e ${{ matrix.tox-environments }} -- --model testing -m '${{ steps.select-tests.outputs.mark_expression }}'"
-        env:
-          CI_PACKED_CHARMS: ${{ needs.build.outputs.charms }}
-
-      - name: Dump logs
-        uses: canonical/charm-logdump-action@main
-        if: failure()
-        with:
-          app: mongodb-k8s
-          model: testing
+    uses: canonical/data-platform-workflows/.github/workflows/integration_test_charm.yaml@v13.1.2
+    with:
+      artifact-prefix: packed-charm-cache-true
+      cloud: microk8s
+      microk8s-snap-channel: 1.29-strict/stable
+      juju-agent-version: 3.1.7
+      libjuju-version-constraint: 3.4.0.0
+    secrets:
+      integration-test: |
+        {
+          "AWS_ACCESS_KEY": "${{ secrets.AWS_ACCESS_KEY }}",
+          "AWS_SECRET_KEY": "${{ secrets.AWS_SECRET_KEY }}",
+          "GCP_ACCESS_KEY": "${{ secrets.GCP_ACCESS_KEY }}",
+          "GCP_SECRET_KEY": "${{ secrets.GCP_SECRET_KEY }}",
+        }
```

```suggestion
        {  "AWS_ACCESS_KEY": "${{ secrets.AWS_ACCESS_KEY }}",
          "AWS_SECRET_KEY": "${{ secrets.AWS_SECRET_KEY }}",
          "GCP_ACCESS_KEY": "${{ secrets.GCP_ACCESS_KEY }}",
          "GCP_SECRET_KEY": "${{ secrets.GCP_SECRET_KEY }}", }
```
(https://github.com/canonical/data-platform-workflows/pull/166)